### PR TITLE
Ensure that unsubscribe link is always saved [MAILPOET-2245]

### PR DESCRIPTION
**Pull request by costasovo**
[MAILPOET-2245](https://mailpoet.atlassian.net/browse/MAILPOET-2245)

### CODE REVIEW

File: lib/Models/NewsletterLink.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
  */
 class NewsletterLink extends Model {
   public static $_table = MP_NEWSLETTER_LINKS_TABLE;
+  const UNSUBSCRIBE_LINK_SHORT_CODE = '[link:subscription_unsubscribe_url]';
```
Comment (pavel-mailpoet) [inline lib/Models/NewsletterLink.php]: If you introduced this constant should we also replace strings to start using it?

File: tests/integration/Newsletter/Links/LinksTest.php [hunk 2]
```diff
@@ -8,6 +8,7 @@
 use MailPoet\Newsletter\Links\Links;
 use MailPoet\Newsletter\Shortcodes\Categories\Link;
 use MailPoet\Router\Router;
+use function MailPoetVendor\Symfony\Component\DependencyInjection\Loader\Configurator\expr;
```
Comment (pavel-mailpoet) [inline tests/integration/Newsletter/Links/LinksTest.php]: Is this used? I can see where.

File: lib/Newsletter/Links/Links.php [hunk 3]
```diff
@@ -170,6 +133,20 @@ static function save(array $links, $newsletter_id, $queue_id) {
     }
   }
 
+  static function ensureUnsubscribeLink(array $processed_links) {
+    if (in_array(
+      NewsletterLink::UNSUBSCRIBE_LINK_SHORT_CODE,
+      \MailPoet\Util\array_column($processed_links, 'link'))
```
Comment (pavel-mailpoet) [inline lib/Newsletter/Links/Links.php]: Why did you use fully qualified namespace here? AFAIK we use `use` everywhere.

File: tests/integration/Newsletter/Links/LinksTest.php (hunk 2 above)
Comment (costasovo) [reply tests/integration/Newsletter/Links/LinksTest.php]: Good catch! This was added by PHP Storm by accident...

**Pull request merged**